packages/ra-core/src/util/Mutation.tsx

```diff
@@ -0,0 +1,83 @@
+import { Component, ReactNode } from 'react';
+import withDataProvider from './withDataProvider';
+
+type DataProviderCallback = (
+    type: string,
```

Comment:
Maybe we can explicitly allow only the mutation types here

Reply:
No, because we want to allow custom dataProvider verbs in general.